Parser/Types/TypeAnalyze.cs

```diff
@@ -86,9 +99,11 @@ public override NType Visit (NWriteStmt w)
       Visit (r.Stmts); r.Condition.Accept (this);
       return Void;
    }
-
+    
    public override NType Visit (NCallStmt c) {
-      throw new NotImplementedException ();
+      if (mSymbols.Find (c.Name.Text) is NVarDecl)
+         Visit (c.Params);
+      throw new ParseException (c.Name, "Unknown variable");

```

Comment:
You should expect to find a NFnDecl, not an NVarDecl here. Did you test this?

Reply:
The compilation is crashing.
Is Fncall different from CallStmt?